test/bigtest/tests/circulation-rules-test.js

```diff
@@ -562,7 +566,102 @@ describe('CirculationRules', () => {
           });
 
           it('the library code should be added to the editor value', () => {
-            expect(circulationRules.editor.value).to.be.equal(`c ${libraries[0].code} `);
+            expect(circulationRules.editor.value).to.be.equal(`c ${replacer(libraries[0].code)} `);
+          });
+        });
+      });
+    });
+  });
+
+  describe('choosing criteria locating criteria - location', () => {
+    const rulesEditor = circulationRules.editor;
+    const editorHints = rulesEditor.hints;
+
+    beforeEach(async function () {
+      await rulesEditor.setValue('s ');
+    });
+
+    it('should display hints with list to select location codes', () => {
+      expect(editorHints.arePresent).to.be.true;
+    });
+
+    it('should display hints with correct header', () => {
+      expect(editorHints.header.isPresent).to.be.true;
+      expect(editorHints.header.text).to.be.equal('Select location');
+    });
+
+    it('should display three hints sections', () => {
+      expect(editorHints.sectionsCount).to.be.equals(4);
```

Comment:
```suggestion
      expect(editorHints.sectionsCount).to.equal(4);
```
It can be slighty shorter. Please handle all similar cases in the file.